### PR TITLE
Crypto-16: Create GraphComponent

### PR DESCRIPTION
**Pull request by vitalyaliskevych**
### Whats News
- Create Graph Component from DetailScreen
### Issues
- [Crypto-16](https://trello.com/c/xEWa4gAt)
### Screenshot
Screenshot 2023-02-14 at 16 09 25

### CODE REVIEW

File: Crypto/Flows/Screens/GraphScreen/GraphView.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        PriceInfo(price: 96.4, day: "05.11")
+    ]
+    
+    static var previews: some View {
+        GraphView(viewModel: .init(coinInfo: coinInfo))
+    }
+}
+
+
+
```
Comment (romansorochak) [inline Crypto/Flows/Screens/GraphScreen/GraphView.swift:67]: remove extra lines in the bottom

File: Crypto/Flows/Screens/GraphScreen/GraphViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+//
+//  Created by Developer on 13.02.2023.
+//
+
+import Foundation
+import SwiftUI
+
+class GraphViewModel: ObservableObject {
+
+    @Published var maxValue: Double = 100
```
Comment (romansorochak) [inline Crypto/Flows/Screens/GraphScreen/GraphViewModel.swift:13]: can it change actually ?

File: Crypto/Flows/Screens/GraphScreen/GraphView.swift [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        }
+    }
+}
+
+private extension GraphView {
+    var createDivider: some View {
+        Color.gray
+            .cornerRadius(.infinity)
+            .frame(width: 310, height: 5)
+            }
```
Comment (romansorochak) [inline Crypto/Flows/Screens/GraphScreen/GraphView.swift:48]: bad alignment

File: Crypto/Flows/Screens/GraphScreen/GraphView.swift (hunk 1 above)
Comment (vitalyaliskevych) [reply Crypto/Flows/Screens/GraphScreen/GraphView.swift:67]: i remove this

File: Crypto/Flows/Screens/GraphScreen/GraphView.swift (hunk 3 above)
Comment (vitalyaliskevych) [reply Crypto/Flows/Screens/GraphScreen/GraphView.swift:48]: i fix

File: Crypto/Flows/Screens/GraphScreen/GraphViewModel.swift (hunk 2 above)
Comment (vitalyaliskevych) [reply Crypto/Flows/Screens/GraphScreen/GraphViewModel.swift:13]: yes it can change, but this value  makes the design more accurate to the given

Comment (vasyl-nadtochii) [reply Crypto/Flows/Screens/GraphScreen/GraphViewModel.swift:13]: @vitalyaliskevych I guess `maxValue` will not be changed anywhere, so we should probably make it `let` and not `@Published`

File: Crypto/Extentions/Array.swift [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+//
+//  Created by Developer on 13.02.2023.
+//
+
+import Foundation
+import SwiftUI
+
+extension Array where Element == PriceInfo {
+    func getColor(index: Int) -> Color {
+            guard index > 0 else {
```
Comment (vasyl-nadtochii) [inline Crypto/Extentions/Array.swift:13]: Bad alignment

File: Crypto/Flows/Screens/GraphScreen/GraphView.swift [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            .frame(width: 330, height: 330)
+            .background(Color.bottomButtonColor.opacity(0.9))
+            .cornerRadius(15)
+        }
+    }
+}
+
+private extension GraphView {
+    var createDivider: some View {
```
Comment (vasyl-nadtochii) [inline Crypto/Flows/Screens/GraphScreen/GraphView.swift:44]: divider

**Review by vasyl-nadtochii — APPROVED** [review]
Please resolve this small comments

File: Crypto/Flows/Screens/GraphScreen/GraphViewModel.swift [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+//
+//  Created by Developer on 13.02.2023.
+//
+
+import Foundation
+import SwiftUI
+
+class GraphViewModel: ObservableObject {
+
+    var maxValue: Double = 100
```
Comment (vasyl-nadtochii) [inline Crypto/Flows/Screens/GraphScreen/GraphViewModel.swift:13]: it can be let

File: Crypto/Flows/Screens/GraphScreen/GraphViewModel.swift [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    var fullBarHeight: Double = 250
```
Comment (vasyl-nadtochii) [inline Crypto/Flows/Screens/GraphScreen/GraphViewModel.swift:14]: it can be let

**Pull request merged**